website/src/main/tut/docs/rules/NoInfer.md

```diff
@@ -1,27 +1,51 @@
 ---
 layout: docs
 title: NoInfer
+tag: rule
 ---
 
 # NoInfer
 
 _Since 0.5.0_
 
-This rule reports errors when the compiler inferred one of the following types:
+This rule reports errors when the compiler infers certain types.
+
+Example:
+
+```scala
+MyCode.scala:7: error: [NoInfer.any] Inferred Any
+  List(1, "")
+            ^
+```
+
+## Configuration
+
+By default the rule reports on the following types:
 
 - `Serializable`
 - `Any`
 - `AnyVal`
 - `Product`
 
-Example:
+To signal when inferring specific types not in the default, something like the below can be created:
 
 ```scala
-MyCode.scala:7: error: [NoInfer.any] Inferred Any
-  List(1, "")
-            ^
+NoInfer.symbols = [
+  "scala.Predef.any2stringadd"
+]
+```
+
+and it would report 
+
+```scala
+MyCode.scala:7: error: [NoInfer.any2stringadd] Inferred any2stringadd
+  def sum[A](a: A, b: String): String = { a + b }
+                                          ^
 ```
 
-## Known limitations}
+**Note:** when a configuration for NoInfer is given it completely overwrites the defaults so they have to be explicitely added to the list of symbols if needed.
```

Comment:
Good to mention this! 👍 

I wonder what we could do make this more intuitive, and make it easier to extend the defaults with custom symbols without overriding. HOCON supports `+=` with reference.conf, so that would be one route. However, I am inclined to migrate all the configuration to JSON in a future release, since there is no good cross-platform (native/js) story for typesafe/config.

Reply:
Supporting `=` and `+=` is good idea.

Reply:
Yes, I agree that sounds like a good idea. I opened https://github.com/scalacenter/scalafix/issues/402 to track this, if you are interested in contributing to it another day :)